selfdrive/ui/replay/logreader.cc

```diff
@@ -38,33 +38,68 @@ bool LogReader::load(const std::string &file) {
     return false;
   }
 
-  auto insertEidx = [&](CameraType type, const cereal::EncodeIndex::Reader &e) {
-    eidx[type][e.getFrameId()] = {e.getSegmentNum(), e.getSegmentId()};
+  std::unordered_map<uint32_t, uint64_t> frame_monotime[MAX_CAMERAS] = {};
+  std::unordered_map<uint32_t, std::pair<const Event *, cereal::EncodeIndex::Reader>> eidx_map[MAX_CAMERAS] = {};
+  auto insertEidx = [&](CameraType type, const cereal::EncodeIndex::Reader &reader, const Event* evt) {
+    eidx_map[type][reader.getFrameId()] = {evt, reader};
   };
 
   kj::ArrayPtr<const capnp::word> words((const capnp::word *)raw_.data(), raw_.size() / sizeof(capnp::word));
   while (words.size() > 0) {
     try {
       std::unique_ptr<Event> evt = std::make_unique<Event>(words);
       switch (evt->which) {
+        case cereal::Event::ROAD_CAMERA_STATE:
```

Comment:
I don't think we need all this extra tracking of the camera state timestamp. In the rare cases that neither EOF or SOF are set we can just use the logMonoTime of the encode packet.